Monika After Story/game/styles.rpy

```diff
@@ -388,3 +388,326 @@ init -10 python in mas_ui:
+            search_kws - search_query splitted using spaces
+
+        OUT:
+            weight as int
+        """
+        ev_prompt = ev.prompt.lower()
+        ev_label = ev.eventlabel.lower()
+        ev_cat_full = " ".join(map(str, ev.category)) if ev.category else ""
+
+        weight = 0
+        base_increment = 2
+        base_modifier = len(search_kws) + 1
+
+        if search_query == ev_prompt or search_query == ev_label:
+            weight += base_increment * base_modifier**8
+
+        elif search_query in ev_prompt:
+            if ev_prompt.startswith(search_query):
+                weight += base_increment * base_modifier**7
+
+            else:
+                weight += base_increment * base_modifier**6
+
+        elif search_query in ev_label:
+            if ev_label.startswith(search_query):
+                weight += base_increment * base_modifier**5
+
+            else:
+                weight += base_increment * base_modifier**4
+
+        else:
+            for search_kw in search_kws:
+                if search_kw in ev_prompt:
+                    weight += base_increment * base_modifier**3
+
+                elif search_kw in ev_label:
+                    weight += base_increment * base_modifier**2
+
+                elif ev_cat_full:
+                    if search_kw in ev.category:
+                        weight += base_increment * base_modifier
+
+                    elif search_kw in ev_cat_full:
+                        weight += base_increment
+
+        return weight
+
+    def _twopane_menu_search_events(search_query):
+        """
+        The actual method that does filtering and searching for the twopane menu.
+        NOTE: won't return more than TWOPANE_MENU_MAX_FLT_ITEMS events
+
+        IN:
+            search_query - search query to filter and sort by
+
+        OUT:
+            list of event objects or None if empty query was given
+        """
+        if not search_query:
+            return None
+
+        search_query = search_query.lower()
+
+        only_pool = False
+        if "#pool" in search_query:
+            search_query = search_query.replace("#pool", "")
+            only_pool = True
+
+        only_random = False
+        if "#random" in search_query:
+            search_query = search_query.replace("#random", "")
+            only_random = True
+
+        only_unseen = False
+        if "#unseen" in search_query:
+            search_query = search_query.replace("#unseen", "")
+            only_unseen = True
+
+        only_seen = False
+        if "#seen" in search_query:
+            search_query = search_query.replace("#seen", "")
+            only_seen = True
+
+        search_query = search_query.strip()
+        search_kws = search_query.split()
+
+        flt_evs = filter(
+            lambda ev: _twopane_menu_filter_events(ev, search_query, search_kws, only_pool, only_random, only_unseen, only_seen),
+            TWOPANE_MENU_SEARCH_DBS
+        )
```

Comment:
FYI, `filter` does not return a list in py3, so the `.sort` call will crash in the future. This also is a scenario where we really don't need to use a lambda function to filter, list comp is fine:
```suggestion
        flt_evs = [
            ev
            for ev in TWOPANE_MENU_SEARCH_DBS
            if _twopane_menu_filter_events(ev, search_query, search_kws, only_pool, only_random, only_unseen, only_seen)
        ]
```